### PR TITLE
Fix errors in manual row and column resize plugins

### PR DESCRIPTION
**Pull request by budnix**
### Context

The PR fixes errors caused by the _ManualRowResize_ and _ManualColumnResize_ plugins when the cell renderer uses the HTML table element to present the value. In that case, mouse movement above the table causes the errors. The fix corrects the code to be more specific about detecting the correct row/column headers. Moreover, the PR changes the CSS selectors to be more specific about what table elements are hidden.

### How has this been tested?

I tested the changes locally, and I covered the fix with two new tests.

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)

### Related issue(s):
1. fixes https://github.com/handsontable/dev-handsontable/issues/1298

### Affected project(s):
- [x] `handsontable`

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)

### CODE REVIEW

File: handsontable/src/plugins/manualColumnResize/manualColumnResize.js [hunk 1]
```diff
@@ -393,7 +393,9 @@ export class ManualColumnResize extends BasePlugin {
    * @returns {boolean}
    */
   checkIfColumnHeader(element) {
-    return !!closest(element, ['THEAD'], this.hot.rootElement);
+    const thead = closest(element, ['THEAD'], this.hot.rootElement);
+
+    return thead === this.hot.view._wt.wtOverlays.topOverlay.clone.wtTable.THEAD;
```
Comment (jansiegel) [inline handsontable/src/plugins/manualColumnResize/manualColumnResize.js:398]: This condition makes the resize handle not appear when the column headers are in the corner overlay (overlapping fixed rows and columns)

File: handsontable/src/plugins/manualRowResize/manualRowResize.js [hunk 2]
```diff
@@ -351,9 +351,9 @@ export class ManualRowResize extends BasePlugin {
    * @returns {boolean}
    */
   checkIfRowHeader(element) {
-    const thElement = closest(element, ['TH'], this.hot.rootElement);
+    const tbody = closest(element, ['TBODY'], this.hot.rootElement);
 
-    return thElement && element.parentNode?.parentNode?.tagName === 'TBODY';
+    return tbody === this.hot.view._wt.wtOverlays.inlineStartOverlay.clone.wtTable.TBODY;
```
Comment (jansiegel) [inline handsontable/src/plugins/manualRowResize/manualRowResize.js:356]: This condition makes the resize handle not appear when the row headers are in the corner overlay (overlapping fixed rows and columns)

**Review by jansiegel — CHANGES_REQUESTED** [review]

File: handsontable/src/plugins/manualColumnResize/manualColumnResize.js (hunk 1 above)
Comment (budnix) [reply handsontable/src/plugins/manualColumnResize/manualColumnResize.js]: Thanks for spotting that. I fixed and covered the change within https://github.com/handsontable/handsontable/pull/10650/commits/e8be061f7b55a3c90e058a284dadd9e3a3c9fa59.

File: handsontable/src/plugins/manualRowResize/manualRowResize.js (hunk 2 above)
Comment (budnix) [reply handsontable/src/plugins/manualRowResize/manualRowResize.js]: Thanks for spotting that. I fixed and covered the change within https://github.com/handsontable/handsontable/pull/10650/commits/e8be061f7b55a3c90e058a284dadd9e3a3c9fa59.

File: handsontable/src/plugins/manualRowResize/manualRowResize.js [hunk 3]
```diff
@@ -351,9 +351,13 @@ export class ManualRowResize extends BasePlugin {
    * @returns {boolean}
    */
   checkIfRowHeader(element) {
-    const thElement = closest(element, ['TH'], this.hot.rootElement);
+    const tbody = closest(element, ['TBODY'], this.hot.rootElement);
+    const { inlineStartOverlay, topInlineStartCornerOverlay } = this.hot.view._wt.wtOverlays;
```
Comment (jansiegel) [inline handsontable/src/plugins/manualRowResize/manualRowResize.js:355]:
```suggestion
    const { inlineStartOverlay, topInlineStartCornerOverlay, bottomInlineStartCornerOverlay } = this.hot.view._wt.wtOverlays;
```

File: handsontable/src/plugins/manualRowResize/manualRowResize.js [hunk 4]
```diff
@@ -351,9 +351,13 @@ export class ManualRowResize extends BasePlugin {
...
 
-    return thElement && element.parentNode?.parentNode?.tagName === 'TBODY';
+    return [
+      inlineStartOverlay.clone.wtTable.TBODY,
+      topInlineStartCornerOverlay.clone.wtTable.TBODY,
+    ].includes(tbody);
```
Comment (jansiegel) [inline handsontable/src/plugins/manualRowResize/manualRowResize.js:360]: This needs an entry for the bottom/start corner as well

```suggestion
    return [
      inlineStartOverlay.clone.wtTable.TBODY,
      topInlineStartCornerOverlay.clone.wtTable.TBODY,
      bottomInlineStartCornerOverlay.clone.wtTable.TBODY
    ].includes(tbody);
```

**Review by jansiegel — CHANGES_REQUESTED** [review]

**Review by jansiegel — APPROVED** [review]

**Pull request merged**